### PR TITLE
Updated knex

### LINKED ISSUE #2600: The knex version we're using has a known bug that prevents testing connections
**Informations**
- **Node.js version**: v10.15.0
- **NPM version**: 6.4.1
- **Strapi version**: 3.0.0-alpha.18
- **Database**: mysql
- **Operating system**: MacOS

**What is the current behavior?**
Strapi uses an old version of knex with a known issue. 
Specifically the issue I'm experiencing is that after connections to the DB have been idle for quite a while, they may be disconnected, but because of https://github.com/tgriesser/knex/issues/2321 (which I've checked the code and the issue is also affecting knex 0.13.0 which is the version we're using), the connection is not actually being tested before it is used.
The comment that shows the actual key of the issue in knex is https://github.com/tgriesser/knex/issues/2321#issuecomment-344409035 and it also suggests the fix: to pass `testOnBorrow: true` as a config to the pool.

Now, because the connection pool options are not very configurable (which is called out in #2030)  we need to change the code in the knex hook to be able to fix it.

Considering that the fix in knex was to set this option as true by default I'd suggest we do the same.

(Alternatively we could upgrade knex, but the later versions use a different connection pool library `tarn` instead of `generic-pool` and that *may* be a breaking change.

**Steps to reproduce the problem**
Bring app up that uses a mysql connection.
Let it sleep for a few hours or kill the connection somehow else
Make a request that need the DB
It has an error and does not recover

**What is the expected behavior?**
It should reconnect to the DB and fulfil the request

**Suggested solutions**
I'll submit a PR, but I believe it should be as simple as setting `testOnBorrow: true` as default on the knex options. Also, TBH, there's no good reason why the pool options are not being set and are configurable wherever we use knex. So I'll include some of that in the PR.

### PR DESCRIPTION
**Pull request by pragone**
My PR is a:
- [ ] 💥 Breaking change
- [x] 🐛 Bug fix #2600 
- [x] 💅 Enhancement
- [ ] 🚀 New feature

Main update on the:
- [ ] Admin
- [ ] Documentation
- [x] Framework
- [x] Plugin

This PR updates knex to the latest version 0.16.2.
It also makes some of the basic pool configuration options available for configuration

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
Nice work man, I'll test this out on MySQL/MariaDB

**Comment by pragone** [comment]
Hey @lauriejim and @derrickmehaffy 
I was trying to test this... actually I was trying to make the tests work. But I wasn't successful. trying to get the tests to work on my machine was painful, and they're broken for me. (I must point, they're broken with the old version of knex, so it's not something caused by the upgrade).
I'm afraid I can't invest the time now to help you fix the tests, but please get that fixed.. I guarantee  that I'll help more if the tests provide a safety net

File: packages/strapi-hook-knex/lib/index.js [hunk 1]
```diff
@@ -106,15 +102,24 @@ module.exports = strapi => {
...
         }, strapi.config.hook.settings.knex);
 
+        options.pool = {
+          min: _.get(connection.options, 'pool.min') || 0,
+          max: _.get(connection.options, 'pool.max') || 10,
+          acquireTimeoutMillis: _.get(connection.options, 'pool.acquireTimeoutMillis') || 2000,
+          createTimeoutMillis: _.get(connection.options, 'pool.createTimeoutMillis') || 2000,
+          idleTimeoutMillis: _.get(connection.options, 'pool.idleTimeoutMillis') || 30000,
+          reapIntervalMillis: _.get(connection.options, 'pool.reapIntervalMillis') || 1000,
+          createRetryIntervalMillis: _.get(connection.options, 'pool.createRetryIntervalMillis') || 200,
```
Comment (lauriejim) [inline packages/strapi-hook-knex/lib/index.js]: For all these lines you can do `_.get(connection.options, 'pool.min', 0)`.

Comment (lauriejim) [reply packages/strapi-hook-knex/lib/index.js]: _.get(object, path, default)

**Comment by derrickmehaffy** [comment]
@lauriejim I'll see if I can test this sometime this week (time permitting) to make sure I don't notice any issues

**Comment by lauriejim** [comment]
Good tank you for your update!
@derrickmehaffy you confirm that's good ?

**Comment by derrickmehaffy** [comment]
@lauriejim I have yet to test this completely yet, give me a few hours (been a busy past few weeks)

**Pull request merged**